### ISSUE TITLE
Combine adjacent links with the same destination

### ISSUE THREAD

**Issue by eons44** [web-infrastructure/doc_architecture#24]
Two adjacent links should not point to the same destination. You almost always want to combine the two links.

For example, it is quite common to link an image and text together like this:
```html
<a href="news.html">
    <img src="photo.jpg" alt="" />
</a>
<a href="news.html">
    News
</a>
```
These adjacent links should be combined like so:
```html
<a href="news.html">
    <img src="photo.jpg" alt="" />
    News
</a>
```
Links to # are ignored, as these are nearly always placeholders for JavaScript, e.g.:
```html
<a href="#" onclick="approve()">
    OK
</a>
<a href="#" onclick="reject()">
    Cancel
</a>
```
Justification

Many users of assistive technology are only able to experience content sequentially - that is, they must step through each part of the page, one part at a time. Often this is a slow process, potentially hampered by a physical impediment.

Stepping through redundant links is time consuming; for example users of a screen reader must hear each link read aloud in turn.

This also simplifies your code.
## Methodology

1. Find all links (<a> tags) on the page.
2. Ignore links pointing to #, as these are usually placeholders for JavaScript.
3. Flag links which are (a) separated only by whitespace and (b) which point to the same absolute URL. 

## Learn more

Read about [Technique H2](https://www.w3.org/TR/WCAG20-TECHS/H2) on how to combine an adjacent image and texts links.
